### ISSUE TITLE
[FEATURE]-CrestronGenericBaseDevice Override OnlineStatusChange

### ISSUE THREAD

**Issue by ngenovese11** [PepperDash/Essentials#141]
**Is your feature request related to a problem? Please describe.**
In the CrestronGenericBaseDevice there is no way to override the method invoked by the OnlineStatusChange - void Hardware_OnlineStatusChange.  There are situations where other things might need to happen when the device comes online, so the easiest way to handle would be to override this method.

**Describe the solution you'd like**
Make the method protected virtual

**Describe alternatives you've considered**
Subscribing to the IsOnline feedback class and handling from there

**Comment by andrew-welker** [comment]
You should be able to subscribe to the actual device's online event in the class that derives from CrestronGenericBaseDevice and do what you need there without overriding or modifying what the base class is doing. Both methods will be called/invoked when the event is triggered from the device. Does that solve your issue?

**Comment by ngenovese11** [comment]
Yes, this was more of a "file this away to think about one day..." type issue.  Thanks!
